File: sources/com/s8/fwks/palm/breadcrumbs/Breadcrumbs.java
package com.s8.fwks.palm.breadcrumbs;

import com.s8.api.web.S8WebFront;
import com.s8.fwks.palm.PalmWebSources;
import com.s8.pkgs.ui.carbide.CarbideSize;
import com.s8.pkgs.ui.carbide.HTML_S8WebNode;


/**
 * 
 * @author pierreconvert
 *
 */
public class Breadcrumbs extends HTML_S8WebNode {
	
	
	
	public final static String WEBPATH = PalmWebSources.ROOT_WEBPATH + "/breadcrumbs/Breadcrumbs";
	
	

	
	/**
	 * 
	 * @param front
	 */
	
	public Breadcrumbs(S8WebFront front) {
		super(front, WEBPATH);
	}
	
	


	/**
	 * 
	 * @param size
	 */
	public void setSize(CarbideSize size) {
		vertex.outbound().setUInt8Field("size", size.code);
	}
	
	
	/**
	 * 
	 * @param items
	 */
	public void setItems(BreadcrumbsNode[] items) {
		vertex.outbound().setObjectListField("items", items);
	}
	
	
	
}

File: web-sources/S8-fwks-palm/breadcrumbs/Breadcrumbs.js

import { S8WebFront } from "/S8-pkgs-ui-carbide/S8WebFront.js";
import { S8Object } from "/S8-api/S8Object.js";
import { BreadcrumbsNode } from "./BreadcrumbsNode.js";




/**
 * 
 */
S8WebFront.CSS_import('/S8-fwks-palm/breadcrumbs/Breadcrumbs.css');


/**
 * 
 */
export class Breadcrumbs extends S8Object {

    /** @type {HTMLDivElement} */
    wrapperNode;
    
    /**
     * 
     */
    constructor(){
        super();
        this.wrapperNode = document.createElement("div");
        this.wrapperNode.classList.add("palm-breadcrumbs");
    }


    /**
     * 
     * @returns 
     */
    getEnvelope(){
        return this.wrapperNode;
    }


    /** @param {BreadcrumbsNode[]} items */
    S8_set_items(items){
       S8WebFront.setS8Elements(this.wrapperNode, items);
    }


    /** render */
    S8_render(){}


    /** dispose */
    S8_dispose(){}

}